apps/boilerplate-front-end/src/components/header/components/OktaAuth.jsx

```diff
@@ -0,0 +1,272 @@
+import { OktaLogo } from '@/assets/svg/SvgIndex'
+import { useOktaAuth } from '@okta/okta-react'
+import { useEffect, useState } from 'react'
+
+const OktaAuth = () => {
+  const { authState, oktaAuth } = useOktaAuth()
+  const handleLogin = () => oktaAuth.signInWithRedirect()
+  // To use for a Logout function
+  // const handleLogout = () => oktaAuth.signOut()
+  const [userEmail, setUserEmail] = useState(null)
+  const [imgLink, setLinkImg] = useState(null)
+  // Get info from okta on user
+  const user = oktaAuth.token
+    .getUserInfo()
+    .then(function (user) {
+      // user has details about the user
+      setUserEmail(user.email)
+    })
+    .catch(function (err) {
+      console.log(err)
+      // handle OAuthError or AuthSdkError (AuthSdkError will be thrown if app is in OAuthCallback state)
+    })
+
+  // Hash function to use email on Gravatar API to have an img
+  function get_gravatar(email, size) {
+    // MD5 (Message-Digest Algorithm) by WebToolkit
+    //
+
+    var MD5 = function (s) {
```

Comment:
is it right that this lives inside the component?

Reply:
I cant move it inside config folder, good point 👍